### PR TITLE
[FSS-3579] Avoid non-dismissible dialog from closing when pressing esc

### PR DESCRIPTION
**Pull request by raullm7**

### CODE REVIEW

File: packages/bpk-component-modal/src/BpkModal.js [hunk 1]
```diff
@@ -52,6 +52,7 @@ const BpkModal = props => {
       onClose={onClose}
       target={target}
       renderTarget={renderTarget}
+      closeOnEscPressed={closeOnScrimClick}
```
Comment (matthewdavidson) [inline packages/bpk-component-modal/src/BpkModal.js]: `closeOnScrimClick` is not he same as `closeOnEscPressed ` - they should be treated as separate props!

Comment (raullm7) [reply packages/bpk-component-modal/src/BpkModal.js]: Wouldn't we want them to share the same logic in this case?

Comment (matthewdavidson) [reply packages/bpk-component-modal/src/BpkModal.js]: We could enable it by default in the Dialog, but the configuration option in the modal should be separate from `closeOnScrimClick`.

It's a surprising side effect for consumers who disable `closeOnScrimClick` and also inadvertently disable that ability to close on esc

Comment (raullm7) [reply packages/bpk-component-modal/src/BpkModal.js]: Cool, I have set it to share the logic with `dismissible` in the dialog component.

**Comment by raullm7** [comment]
I have implemented all the changes.

File: changelog.md [hunk 2]
```diff
@@ -9,6 +9,8 @@ _Nothing yet._
 **Added:**
 - react-native-bpk-component-button: 4.2.16 => 4.3.0
   - New `iconAlignment` prop allows icons inside buttons to be leading or trailing.
+- bpk-react-utils:
```
Comment (matthewdavidson) [inline changelog.md]: This includes `bpk-component-modal` too.

Also there needs to be an entry for `bpk-component-dialog`:
```
**Fixed:**
- bpk-component-dialog:
  - Setting `dissmissable` to false now correctly disables the close on `esc` key press
```

File: packages/bpk-component-modal/src/BpkModal.js [hunk 3]
```diff
@@ -71,13 +73,15 @@ BpkModal.propTypes = {
   target: PropTypes.oneOfType([PropTypes.func, PropTypes.element]),
   fullScreenOnMobile: PropTypes.bool,
   closeOnScrimClick: PropTypes.bool,
+  closeOnEscPressed: PropTypes.bool,
```
Comment (matthewdavidson) [inline packages/bpk-component-modal/src/BpkModal.js]: This is a new prop that has to be added to the modals readme.

File: packages/bpk-react-utils/src/Portal.js [hunk 4]
```diff
@@ -264,6 +268,7 @@ Portal.propTypes = {
   renderTarget: PropTypes.func,
   target: PropTypes.oneOfType([PropTypes.func, PropTypes.node]),
   targetRef: PropTypes.func,
+  closeOnEscPressed: PropTypes.bool,
```
Comment (matthewdavidson) [inline packages/bpk-react-utils/src/Portal.js]: This is a new prop that has to be added to the Portals readme.

File: changelog.md (hunk 2 above)
Comment (shaundon) [reply changelog.md]: Bearing in mind that `bpk-component-dialog` is still private and pre-1.0, do we still need to do this?

Comment (matthewdavidson) [reply changelog.md]: Nope you are correct! apologies

**Comment by raullm7** [comment]
Cool, I have just changed the changelog

**Pull request merged**